docs/new-editor.md

```diff
@@ -0,0 +1,163 @@
+# Integrating a new editor
+
+Before writing a new editor client, first check if someone else has managed to
+integrate metals with your favorite text editor.
+
+* [VS Code](https://github.com/scalameta/metals/blob/master/vscode-extension/src/extension.ts),
+  maintained in this repo
+* [Atom](https://github.com/laughedelic/atom-ide-scala), maintained by
+  [@laughedelic](https://github.com/laughedelic)
+* [Emacs](https://github.com/rossabaker/lsp-scala), maintained by
+  [@rossabaker](https://github.com/rossabaker)
+* Others, see [#217](https://github.com/scalameta/metals/issues/217). Please
+  open an issue or ask on [gitter](https://gitter.im/scalameta/scalameta) if you
+  want to create a new editor client.
+
+To integrate metals with a new editor, a couple of things should be kept in mind
+
+<!-- TOC depthFrom:2 depthTo:2 -->
```

Comment:
just curious, what is this meta comment for?

Reply:
The TOC is automatically generated for me on file save by https://github.com/AlanWalk/Markdown-TOC

Reply:
ahh, sweet.